### PR TITLE
refactored code to separate step of paymaster response

### PR DESCRIPTION
**Pull request by leekt**

### CODE REVIEW

File: packages/sdk/src/BaseAccountAPI.ts [hunk 1]
```diff
@@ -397,58 +402,53 @@ export abstract class BaseAccountAPI {
...
+            ...userOp,
             callData: await this.encodeExecuteBatch([
               await this.paymasterAPI.createGasTokenApprovalRequest(this.provider),
               mainCall
             ]),
             callGasLimit: await this.provider.estimateGas({
               from: this.entryPointAddress,
               to: this.getAccountAddress(),
-              data: callData
+              data: userOp.callData
```
Comment (zobeirhamid) [inline packages/sdk/src/BaseAccountAPI.ts:413]: Are you sure this is correct? It previously took the encoded callData.

File: packages/sdk/src/BaseAccountAPI.ts [hunk 2]
```diff
@@ -378,7 +378,12 @@ export abstract class BaseAccountAPI {
     partialUserOp.preVerificationGas = this.getPreVerificationGas(partialUserOp)
     // this is needed for the 0.6 StackUp bundlers
     partialUserOp.paymasterAndData = '0x'
+    return {
+      ...partialUserOp,
+    }
+  }
```
Comment (zobeirhamid) [inline packages/sdk/src/BaseAccountAPI.ts:384]: Can't you just return partialUserOp, why creating a shallow copy?

File: packages/sdk/src/session/SessionSigner.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ import { BaseAccountAPI, ExecuteType } from '../BaseAccountAPI'
 import * as constants from '../constants'
 
 // Deterministically deployed against 0.6 EntryPoint
-export const DEFAULT_SESSION_KEY_PLUGIN = '0x6E2631aF80bF7a9cEE83F590eE496bCc2E40626D'
+export const DEFAULT_SESSION_KEY_PLUGIN = '0xA08c304A85b8C73c5847b08DBcfD3bF7E6CC607A'
```
Comment (zobeirhamid) [inline packages/sdk/src/session/SessionSigner.ts:23]: Will this work with old sessionKeys?

**Review by zobeirhamid — COMMENTED** [review]
The biggest question is, will old session key work with this or not? This will determine if it's a minor upgrade or a patch. And we would need to inform Synquote

**Comment by zobeirhamid** [comment]
Just tried using this, the paymaster is returning: AA23 reverted (or OOG)

**Comment by leekt** [comment]
> The biggest question is, will old session key work with this or not? This will determine if it's a minor upgrade or a patch. And we would need to inform Synquote

Hmmm no it won't it's because we changed address

File: packages/sdk/src/BaseAccountAPI.ts (hunk 2 above)
Comment (leekt) [reply packages/sdk/src/BaseAccountAPI.ts:384]: kk i'll return partialUserOp

File: packages/sdk/src/BaseAccountAPI.ts (hunk 1 above)
Comment (leekt) [reply packages/sdk/src/BaseAccountAPI.ts:413]: yup previous callData(encoded one) is updated to userOp in createUnsignedUserOp function and that is passed on to this function afterwards

**Pull request merged**